docs/supported-tools.md

```diff
@@ -0,0 +1,271 @@
+
+  <!--region File exclusions-->
+  <exclude-pattern>.*\.kt</exclude-pattern>
+  <!--endregion-->
+
+  ...
+
+</ruleset>
+```
+
+## Findbugs
+[Findbugs](http://findbugs.sourceforge.net/) is a static analysis tool that looks for potential bugs in Java code. It does not support Kotlin.
+It can be used in both pure Java, and Android Java projects. It then only makes sense to have Findbugs enabled if you have Java code in your project.
+The plugin only runs Findbugs on projects that contain the Java or the Android plugin.
+
+### Configure Findbugs
+Enabling and configuring Findbugs for a project is done through the `findbugs` closure:
+
+```gradle
+findbugs {
+    toolVersion // A string, most likely '3.0.1' — the latest Findbugs release (for a long time)
+    exclude // A fileTree, such as project.fileTree('src/test/java') to exclude Java unit tests
+    excludeFilter // A file containing the Findbugs exclusions, e.g., teamPropsFile('static-analysis/findbugs-excludes.xml')
+    includeVariants { variant -> ... } // A closure to determine which variants (for Android) to include
+}
+```
+
+You can have multiple `exclude` statements.
+
+For more informations about Findbugs rules, refer to the [official website](http://findbugs.sourceforge.net/bugDescriptions.html).
+
+### Findbugs in mixed-language projects
+If your project mixes Java and Kotlin code, you most likely want to have an exclusion in place for all `*.kt` files. You can use the `exclude`
+in the configuration closure, or you can do so by adding a suppression to the filter file:
+
+`findbugs-excludes.xml`
+```xml
+<FindBugsFilter>
+
+  ...
+
+  <Match>
+    <Source name="~.*\.kt" />
+  </Match>
+
+  ...
+
+</FindBugsFilter>
+```
+
+## Detekt
+[Detekt](https://github.com/arturbosch/detekt) is a static analysis tool that looks for potential bugs and style violations in Kotlin code. It
+does not support Java. It can be used in both pure Kotlin, and Android Kotlin projects. It then only makes sense to have Detekt enabled if you
+have Kotlin code in your project. The plugin only runs Detekt on projects that contain the Kotlin or the Kotlin-Android plugin.
+
+### IMPORTANT: setup Detekt
+Unlike the other tools, the plugin **won't automatically add Detekt** to your project. If you forget to do it, the plugin will fail the build
+with an error like:
+
+```
+The detekt plugin is configured but not applied. Please apply the plugin in your build script.
+For more information see https://github.com/arturbosch/detekt.
+```
+
+In order to use Detekt in your project, you need to:
+
+ 1. Add this statement to your root `build.gradle` project (change the version according to your needs):
+    ```gradle
+    plugins {
+        id 'io.gitlab.arturbosch.detekt' version '1.0.0.RC6-2'
+        // ...
+    }
+    ```
+ 2. Add this statement to each Kotlin project's `build.gradle`s:
+    ```gradle
+    plugins {
+        id 'io.gitlab.arturbosch.detekt'
+        // ...
+    }
+    ```
+
+### Configure Detekt
```

Comment:
Should we add that we suggest to disable thresholds in the `detekt` config file?

Reply:
Good point. Since we cannot automatically do this, it is really important to mention.

Reply:
Yes! I totally forgot about it!

Reply:
a38c70b748b653e46cae764d2025c26faa061f77